components/conformance/README.md

```diff
@@ -0,0 +1,289 @@
+  - `html` - HTML report file
+  - `json` - JSON format for programmatic consumption
+
+- `--output-file <path>` - Path to write output file (for non-console reporters)
+
+#### Test Selection
+
+- `--category <categories>` - Run only specific test categories (comma-separated)
+  - Example: `--category labels,nodeinfo`
+  - Available categories:
+    - `labels` - Label endpoint tests
+    - `datasets` - Dataset endpoint tests (when implemented)
+    - `nodeinfo` - NodeInfo endpoint tests
+    - `json-ld` - JSON-LD format validation across all endpoints
+    - `http` - HTTP protocol compliance
+    - `security` - Security-related tests
+
+- `--exclude-category <categories>` - Exclude specific test categories (comma-separated)
+
+#### Other Options
+
+- `--help` - Display help information
+- `--version` - Display version information
+- `--verbose` - Enable verbose output
+- `--no-color` - Disable colored output
+
+### Examples
+
+#### CI/CD Integration
+
+```bash
+# Generate JUnit report for CI systems
+fires-conformance \
+  --url https://staging.fires.example.org \
+  --reporter junit \
+  --output-file test-results.xml
+```
+
+#### Selective Testing
+
+```bash
+# Test only Labels endpoints
+fires-conformance --url https://fires.example.org --category labels
+
+# Test everything except security tests
+fires-conformance --url https://fires.example.org --exclude-category security
+
+# Test JSON-LD and HTTP compliance
+fires-conformance --url https://fires.example.org --category json-ld,http
+```
+
+#### Docker Examples
+
+```bash
+# Test local development server
+docker run --rm --network host \
+  ghcr.io/fedimod/fires-conformance \
+  --url http://localhost:3333
+
+# Generate JUnit report for CI
+docker run --rm -v $(pwd):/output \
+  ghcr.io/fedimod/fires-conformance \
+  --url https://fires.example.org \
+  --reporter junit \
+  --output-file /output/results.xml
+
+# Test with verbose output
+docker run --rm \
+  ghcr.io/fedimod/fires-conformance \
+  --url https://fires.example.org \
+  --verbose
+```
+
+## What Gets Tested
+
+The conformance suite validates:
+
+### Labels Endpoint
+- Collection endpoint (`/labels`) returns valid JSON-LD
+- Individual label endpoints (`/labels/:id`) return valid JSON-LD
+- Content negotiation (HTML vs JSON-LD based on Accept header)
+- Pagination behavior
+- Label structure and required fields
+- Linked data context validity
+
+### NodeInfo Endpoint
+- Well-known discovery endpoint (`/.well-known/nodeinfo`)
+- NodeInfo 2.1 endpoint structure
+- Required metadata fields
+- Protocol identification
+
+### Datasets Endpoint (when implemented)
+- Dataset collection endpoints
+- Individual dataset retrieval
+- Resumable data transfer
+- Change tracking
+
+### HTTP Compliance
+- Appropriate status codes
+- Content-Type headers
+- CORS headers (if applicable)
+- Rate limiting behavior
+
+### Security
+- HTTPS availability (for production servers)
+- No sensitive data leakage
+- Proper authentication rejection
```

Comment:
These are out of scope, if the server under test is on `https://` then yes, ensure there's not a downgrade to http, but otherwise it's not relevant to test, there's plenty of other tools for confirming correct https setup.